examples/text_to_image/train_text_to_image_lora_sdxl.py

```diff
@@ -1187,6 +1198,9 @@ def compute_time_ids(original_size, crops_coords_top_left):
         torch.cuda.empty_cache()
 
         # Final inference
+        # Make sure vae.dtype is consistent with the unet.dtype
+        if args.mixed_precision == "fp16":
+            vae.to(weight_dtype)
```

Comment:
This is not needed in my opinion. We already set the `torch_dtype` in the pipeline during loading it.

Reply:
If no other `pretrained_vae_model_name_or_path` is set, then the vae is set to float32,
```
if args.pretrained_vae_model_name_or_path is None:
        vae.to(accelerator.device, dtype=torch.float32)
```
and the pipeline here does not reload vae. So `vae.dtype` `float32` != `unet.dtype` `fp16`, which in my tests causes `RuntimeError: Input type (c10::Half) and bias type (float) should be the same`

Reply:
Ah okay got it!